File: elixir/series/series.exs
defmodule StringSeries do
  @doc """
  Given a string `s` and a positive integer `size`, return all substrings
  of that size. If `size` is greater than the length of `s`, or less than 1,
  return an empty list.
  """
  @spec slices(s :: String.t(), size :: integer) :: list(String.t())
  def slices(s, size) when size > 0 do
    String.split(s, "", trim: true)
    |> Enum.chunk_every(size, 1, :discard)
    |> Enum.map(&Enum.join/1)
  end

  def slices(_s, _size), do: []
end

File: ruby/series/series.rb
class Series

  def initialize(serie)
    @serie = serie.split('')
  end

  def slices(amount)
    raise ArgumentError if amount > @serie.size
    @serie.each_with_index.map do |e, i|
      @serie[i..i+amount-1].join
    end.reject do |serie|
      serie.size != amount
    end
  end
end
